### PR TITLE
Mixed improvements

### PR DESCRIPTION
**Pull request by kretep**
- Fix typos, documentation
- Validation for date format
- Validate on write
- Update workflows

### CODE REVIEW

File: src/tsdf/tsdfmetadata.py [hunk 1]
```diff
@@ -48,11 +50,11 @@ class TSDFMetadata:
 
     file_dir_path: str
     """ A reference to the directory path, so we don't need it again when reading associated binary files. """
-    metadata_file_name: str
+    metadata_file_name: str #TODO: do we need this?? / is it used?
```
Comment (vedran-kasalica) [inline src/tsdf/tsdfmetadata.py:53]: This is only used when we want to override the existing metadata file by updating the corresponding tsdf object, but that might be unnecessary.

Comment (kretep) [reply src/tsdf/tsdfmetadata.py:53]: So when saving existing metadata at a new location, it is not needed to update this, right?

Comment (vedran-kasalica) [reply src/tsdf/tsdfmetadata.py:53]: As far as I know, when saving to a new location the location is expected to be explicitly specified when writing to the disc (this is more of an implicit location tag)

File: src/tsdf/write_binary.py [hunk 2]
```diff
@@ -19,12 +19,10 @@ def write_dataframe_to_binaries(
     """
     Save binary file based on the provided pandas DataFrame.
 
-    :param file_dir: path to the directory where the file will be saved.
-    :param file_name: name of the file to be saved.
-    :param data: pandas DataFrame containing the data.
-    :param metadata: dictionary containing the metadata.
-
-    :return: TSDFMetadata object.
+    :param file_dir:    path to the directory where the file will be saved.
```
Comment (vedran-kasalica) [inline src/tsdf/write_binary.py:22]: We could at one point convert all the documentation to NumPy style (we started it for the `dbpd`), just not to forget.

**Review by vedran-kasalica — APPROVED** [review]
The PR looks good. The validation on the formatting was overdue, and the improved testing is quite useful!

**Pull request merged**